### PR TITLE
Reenable sensor_msgs test

### PR DESCRIPTION
**Pull request by gleichdick**
This was separated from #368

### CODE REVIEW

**Comment by clalancette** [comment]
Thanks for splitting this out.  CI:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14687/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9439/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12352/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14830/)

**Comment by clalancette** [comment]
@gleichdick FYI, please take a look at 7eb81ba and let me know what you think.  I did a few additional cleanups in there.

**Review by clalancette — APPROVED** [review]

**Comment by gleichdick** [comment]
Thanks, your cleanup commit looks good!

**Comment by clalancette** [comment]
OK, Windows was complaining about stuffing doubles into floats.  I don't have a great solution for that, so in 283cc4b, I ended up just doing a cast.  That can lead to subtle problems in some situations, I guess, but I don't know what else to do there.  Also, this is pre-existing behavior; we are just seeing it now because we added tests.

Here's another round of CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14688/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9440/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12353/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14831/)

**Comment by clalancette** [comment]
@ahcorde This all looks good to me, but since I've made some non-trivial contributions here, I'd appreciate a second review from you.

File: tf2_sensor_msgs/package.xml
Comment (ahcorde) [inline tf2_sensor_msgs/package.xml]: remove

File: tf2_sensor_msgs/package.xml
Comment (ahcorde) [inline tf2_sensor_msgs/package.xml]: Remove

File: tf2_sensor_msgs/test/test_tf2_sensor_msgs.cpp
Comment (ahcorde) [inline tf2_sensor_msgs/test/test_tf2_sensor_msgs.cpp]: this looks wierd.
```suggestion
    cloud, "B", tf2::durationFromSec(2.0));
```

File: tf2_sensor_msgs/test/test_tf2_sensor_msgs.cpp
Comment (ahcorde) [inline tf2_sensor_msgs/test/test_tf2_sensor_msgs.cpp]:
```suggestion
  sensor_msgs::msg::PointCloud2 cloud_advanced = tf_buffer->transform(
    cloud, "B", tf2::timeFromSec(2.0), "A", tf2::durationFromSec(3.0));
```

File: tf2_sensor_msgs/test/test_tf2_sensor_msgs.cpp
Comment (ahcorde) [inline tf2_sensor_msgs/test/test_tf2_sensor_msgs.cpp]: Did you try this?

```suggestion
  t.header.stamp.sec = rclcpp::Time(2, 0);
```

File: tf2_sensor_msgs/test/test_tf2_sensor_msgs.cpp
Comment (ahcorde) [inline tf2_sensor_msgs/test/test_tf2_sensor_msgs.cpp]: same as below

**Review by ahcorde — CHANGES_REQUESTED** [review]

**Comment by gleichdick** [comment]
Btw, closes #365

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14691/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9443/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12356/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14834/)

File: tf2_sensor_msgs/package.xml
Comment (ahcorde) [inline tf2_sensor_msgs/package.xml:27]: I still think we should remove this. It's a ros1 dependency

**Review by ahcorde — CHANGES_REQUESTED** [review]
small nit

File: tf2_sensor_msgs/package.xml
Comment (ahcorde) [reply tf2_sensor_msgs/package.xml:27]: @gleichdick I will relaunch CI when this line will be removed

Comment (gleichdick) [reply tf2_sensor_msgs/package.xml:27]: I removed them, furhtermore some more flake8 issues are resolved which the CI found. Maybe we should add the flake8 modules mentioned [here](https://docs.ros.org/en/crystal/Installation/Linux-Development-Setup.html#install-development-tools-and-ros-tools) to the CI of this project (not the big one)?

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14693/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9445/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12358/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14837/)

File: tf2_sensor_msgs/package.xml
Comment (clalancette) [reply tf2_sensor_msgs/package.xml:27]: While I agree we should eventually do that, I don't think we should do that here.  The next step for this package in my opinion is to split it up into separate `tf2_sensor_msgs` and `tf2_sensor_msgs_py` packages, and actually get the Python stuff working (which it currently isn't).  At that point, it would make sense to enable the automated flake8 tests.

**Review by ahcorde — APPROVED** [review]

**Comment by clalancette** [comment]
All right, we have two approvals and green CI.  I'm going to merge this one, thanks for splitting it out @gleichdick .

**Pull request merged**
